src/com/yetanalytics/lrs_admin_ui/views/update_password.cljs

```diff
@@ -0,0 +1,70 @@
+(ns com.yetanalytics.lrs-admin-ui.views.update-password
+  (:require [reagent.core :as r]
+            [re-frame.core :refer [subscribe dispatch]]
+            [com.yetanalytics.lrs-admin-ui.functions :as fns]
+            [com.yetanalytics.lrs-admin-ui.functions.copy :refer [copy-text]]
+            [com.yetanalytics.lrs-admin-ui.functions.password :as pass]
+            [com.yetanalytics.lrs-admin-ui.input :refer [p-min-len]]
+            [goog.string :refer [format]]
+            goog.string.format))
+
+(defn form []
+  (let [hide-pass (r/atom true)]
+    (fn []
+      (let [new-password @(subscribe [:update-password/new-password])]
+        [:div {:class "create-account-inputs"}
+         [:div {:class "row"}
+          [:label {:for "old-password-input"} "Old Password:"]
+          [:input {:value @(subscribe [:update-password/old-password])
+                   :class "new-account round"
+                   :id "old-password-input"
+                   :type "password"
+                   :on-change #(dispatch [:update-password/set-old-password
+                                          (fns/ps-event-val %)])}]]
+         [:div {:class "row pt-2"}
+          [:label {:for "new-password-input"} "New Password:"]
+          [:input (cond-> {:value new-password
+                           :class "new-account round"
+                           :id "new-password-input"
+                           :on-change #(dispatch [:update-password/set-new-password
+                                                  (fns/ps-event-val %)])}
+                    @hide-pass (merge {:type "password"}))]
+          [:ul {:class "action-icon-list"}
+           [:li
+            [:a {:href "#!",
+                 :class "icon-secret pointer"
+                 :on-click #(swap! hide-pass not)}
+             (str (cond
+                    @hide-pass "Show"
+                    :else "Hide"))]]
+           [:li
+            [copy-text
+             {:text new-password
+              :on-copy #(dispatch [:notification/notify false
+                                   "Copied New Password!"])}
+             [:a {:class "icon-copy pointer"} "Copy"]]]]]
+         [:span {:class "password-note"}
+          (format "New password must be different from old password and be %d or more characters and contain uppercase, lowercase, numbers, and special characters (%s). Be sure to note or copy the new password as it will not be accessible after creation."
+                  p-min-len
+                  (apply str pass/special-chars))]
```

Comment:
Please extract the string format op into its own helper function.

Reply:
I think hyper-extracting simple things out into helpers is worse, it makes it take longer to see what is going on. As I've said in other comments, this is directly carried over from the other password form.